### ISSUE TITLE
twisted.python.filepath.FilePath.islink will never return True

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1773]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1773|
|Type|defect|
|Created|2006-06-01 14:40:15Z|

This is clearly a defect, since some files are, in fact, links.

<details><summary>Searchable metadata</summary>

```
trac-id__1773 1773
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1149172815000000 1149172815000000
changetime__1149380796000000 1149380796000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set status to `closed`|
|-|-|

(In [17040]) Sometimes, filesystem paths are links.

Author: glyph

Reviewer: itamarst

Fixes [#1773](https://github.com/twisted/twisted/issues/1773)

This change fixes it so that sometimes FilePath.islink() will return true when
the destination is in fact a link.  Previously, 'stat' results were cached, and
stat follows symbolic links.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
